firebase-firestore/src/main/java/com/google/firebase/firestore/util/Preconditions.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.util;
+
+import static com.google.firebase.firestore.util.Strings.lenientFormat;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class Preconditions {
```

Comment:
Please mention the source of this file somewhere (preferably a GitHub link to the Guava sources)

Reply:
Done.